extensions/notebook/src/book/bookTreeItem.ts

```diff
@@ -0,0 +1,24 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+export class Book extends vscode.TreeItem {
+
+	constructor(
+		public readonly title: string,
+		public readonly root: string,
+		public readonly toc: any[],
+		public readonly collapsibleState: vscode.TreeItemCollapsibleState,
+		public uri?: string,
```

Comment:
The first BookTreeItem (the top of the book directory) doesn't have a uri since it's just the name of the book. For example, 'SQL Server ML Services' doesn't have a uri but all of its children (the actual notebooks) will have a uri
![Capture](https://user-images.githubusercontent.com/23462877/59882680-c3f46880-9367-11e9-99c6-ce1e6a567023.PNG)